File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.java
class Solution {
    public int maxDifference(String s) {
        HashMap<Character,Integer> mp=new HashMap<>();
        for(char ch:s.toCharArray()){
            mp.put(ch,mp.getOrDefault(ch,0)+1);
        }
        int maxOdd=Integer.MIN_VALUE;
        int minEven=Integer.MAX_VALUE;
        for(Map.Entry<Character,Integer> entry:mp.entrySet()){
            if(entry.getValue()%2==0){
                minEven=Math.min(minEven,entry.getValue());
            }
            else{
                maxOdd=Math.max(maxOdd,entry.getValue());
            }
        }
        return maxOdd-minEven;
    }
}

File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.cpp
class Solution {
public:
    int maxDifference(string s) {
        map<int,int> mp;
        for(char ch:s){
            mp[ch]++;
        }
        int minEven=INT_MAX;
        int maxOdd=INT_MIN;
        for(auto entry:mp){
            if(entry.second%2==0){
                minEven=min(minEven,entry.second);
            }
            else{
                maxOdd=max(maxOdd,entry.second);
            }
        }
        return maxOdd-minEven;
    }
};